src/main/java/frc/robot/systems/BallHandlingFSM.java

```diff
@@ -86,7 +89,7 @@ public FSMState getCurrentState() {
 	public void reset() {
 		currentState = FSMState.IDLE;
 
-		pushCommandTime = -1;
+		isSolenoidExtended = false;
```

Comment:
You can use pushSolenoid.get() to fetch the solenoid state, no need for separate state var. You can fix this in next patch set.